### ISSUE TITLE
Feature request: add_author_via_gh

### ISSUE THREAD

**Issue by maelle** [r-lib/desc#71]
It might be be cool to have a function that'd take a GitHub username as input and add an author based on info returned from GitHub API (name, email address if available). It'd mean adding `gh` in Suggests (it's already sort of suggested via `whoami`). I'd have used this function right now, I'm about to do some copy-pasting. 🙀

**Comment by gaborcsardi** [comment]
That's a good idea. OTOH it is really hard to parse given name and family name out of a full name...

**Comment by maelle** [comment]
Right.

* One could suggest https://cran.r-project.org/web/packages/humaniformat/index.html

* But doesn't `add_me()` already parse that from the full name returned by `whoami` anyway?

**Comment by maelle** [comment]
:eyes: https://github.com/r-lib/desc/blob/0ae001718961e15a63c6ae3be50fb6cbd75bdfe3/R/authors-at-r.R#L269 😉

**Comment by gaborcsardi** [comment]
Good point. It should  actually use `as.person`...

**Comment by maelle** [comment]
So should I do a PR sometimes this week to:

* create a helper function for parsing full names using `as.person()`

* use that helper function in `add_me()`

* add a function called `add_author_via_gh()`?

😺

**Comment by gaborcsardi** [comment]
Yeah. I think 
* `add_author` could  take a `person` object, alternatively, maybe.
*  then add me, would just call `as.person()`
*  and `add_gh_author()` would do the same

**Comment by gaborcsardi** [comment]
Actually, forget the  previous comment.

**Comment by gaborcsardi** [comment]
Yeah, your original plan is good.

**Comment by maelle** [comment]
Btw I'm wondering how one could make sure none of the decisions made in `add_me()`&co isn't culturally biased against some name structures. Names will have to fit in the given/family structure to be accepted on CRAN but the transition from full name to given/family shouldn't assume too much about the full name. 🤔  Not sure where to get advice on that.

**Comment by maelle** [comment]
For a name like "Ada Colau Ballano" (where Ada is the first name and Colau Ballano the last names) neither `humaniformat` nor `as.person()` would handle correctly the double last name. For now I'll use `as.person()`.

**Comment by llrs** [comment]
I  have used `person(given = "Ada", family = "Colau Ballano")` but some people use hypens in English communities/publications `person(given = "Ada", family = "Colau-Ballano")`. In any case the addition of the ORCID in your previous PR makes it easier to know to whom it refers.
HTH
